spec/detour_spec.lua

```diff
@@ -57,6 +58,48 @@ describe("detour", function ()
         assert.same(#vim.api.nvim_list_wins(), 2)
     end)
 
+    it("create nested popup over non-Detour popup", function ()
+        local base_buffer = vim.api.nvim_get_current_buf()
+        local non_detour_popup = vim.api.nvim_open_win(
+            vim.api.nvim_win_get_buf(0),
+            true,
+            {relative='win', width=12, height=3, bufpos={100,10}}
+        )
+
+        -- this is all copied from the test case below; see the note which follows this function for
+        -- a note regarding how these tests could be cleaned up.
+        local parent_popup = vim.api.nvim_get_current_win()
+        local parent_buffer = vim.api.nvim_get_current_buf()
+        assert.True(util.is_floating(parent_popup))
+        local parent_config = vim.api.nvim_win_get_config(parent_popup)
+
+        detour.Detour()
+        local child_popup = vim.api.nvim_get_current_win()
+        assert.True(util.is_floating(child_popup))
+        local child_config = vim.api.nvim_win_get_config(child_popup)
+        local child_buffer = vim.api.nvim_get_current_buf()
+        -- The nested popup should always be on top of the parent popup
+        assert.True(child_config.zindex > parent_config.zindex)
+
+        assert.same(#vim.api.nvim_list_wins(), 3)
+        assert.same(base_buffer, parent_buffer)
+        assert.same(parent_buffer, child_buffer)
+
+        vim.cmd.quit()
+        assert.same(#vim.api.nvim_list_wins(), 2)
+        vim.cmd.quit()
+        assert.same(#vim.api.nvim_list_wins(), 1)
+    end)
+    -- TODO: technically, either 
+    --   1) all test cases involving nested popups need to be tested with both
+    --      nested Detour popups and nested windows generally; or
+    --   2) we must ensure that detour.Detour() behaves identically when nesting over either of the
+    --      objects just mentioned.
+    -- Approach (2) is clearly superior, but it's not clear (to neilvyas) how to accomplish that;
+    -- as such, perhaps the better approach is to abstract all nesting-related coverage into a
+    -- function which accepts a "popup constructor" and produces test cases, then invoke this
+    -- function twice, once with a constructor for detour.Detour() and another for general popups.
```

Comment:
I think this text goes beyond just a simple TODO but should instead be a conversation on a github issue (where the conversation can be tracked). Could you move this message to a github issue?

Reply:
aye aye!

Reply:
moved the discussion to #25